packages/proto-rpc/src/ServerRegistry.ts

```diff
@@ -0,0 +1,134 @@
+import { RpcMessage } from './proto/ProtoRpc'
+import EventEmitter = require('events')
+import { MethodInfo, RpcMetadata, RpcStatus, ServerCallContext } from '@protobuf-ts/runtime-rpc'
+import { BinaryReadOptions, BinaryWriteOptions } from '@protobuf-ts/runtime'
+import { promiseTimeout } from './common'
+import { Err } from './errors'
+import UnknownRpcMethod = Err.UnknownRpcMethod
+import { Logger } from './Logger'
+import { ConversionWrappers } from './ConversionWrappers'
+import { ProtoRpcOptions } from './ClientTransport'
+import { Empty } from './proto/google/protobuf/empty'
+
+export enum ServerRegistryEvent {
+    RPC_RESPONSE = 'streamr:dht-transport:server:response-new',
+    RPC_REQUEST = 'streamr:dht-transport:server:request-new',
+}
+
+export interface ServerRegistry {
+    on(event: ServerRegistryEvent.RPC_RESPONSE, listener: (rpcMessage: RpcMessage) => void): this
+    on(event: ServerRegistryEvent.RPC_REQUEST, listener: (rpcMessage: RpcMessage) => void): this
+}
+
+export interface Parser<Target> { fromBinary: (data: Uint8Array, options?: Partial<BinaryReadOptions>) => Target }
+export interface Serializer<Target> { toBinary: (message: Target, options?: Partial<BinaryWriteOptions>) => Uint8Array }
+
+export type RegisteredMethod = (request: Uint8Array, callContext: CallContext) => Promise<Uint8Array>
+export type RegisteredNotification = (request: Uint8Array, callContext: CallContext) => Promise<Empty>
```

Comment:
Types are only used within this module
```suggestion
type RegisteredMethod = (request: Uint8Array, callContext: CallContext) => Promise<Uint8Array>
type RegisteredNotification = (request: Uint8Array, callContext: CallContext) => Promise<Empty>
```